### PR TITLE
camltc-0.9.4

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

File: packages/camltc/camltc.0.9.4/url [hunk 1]
```diff
@@ -0,0 +1,2 @@
+git: "https://github.com/toolslive/camltc.git#camltc-0.9.4"
```
Comment (avsm) [inline packages/camltc/camltc.0.9.4/url]: would it be possible to get a stable archive to merge into opam-repository, rather than a git tag?

Comment (toolslive) [inline packages/camltc/camltc.0.9.4/url]: Do you prefer a commit hash, or do you want an url that points to an archive combined with a checkum ?

**Comment by toolslive** [comment]
can someone give travis a kick ?

**Comment by samoht** [comment]
restarted

**Comment by toolslive** [comment]
It seems the failure is tied to opam version 1.2.0. I guess it doesn't understand the commit hash (alas, travis doesn't show what's happening there).

I would normally just make it unavailable for this particular opam, but the constraint language for the opam file doesn't seem to let me:
- `opam-version` needs to be "1.2"
- `available` doesn't have something for the `opam-version` (according to the manual, you can just specify `os` and `ocaml-version`)

now what?

**Comment by yallop** [comment]
@toolslive, you can include constraints on `opam-version` (e.g. `opam-version > "1.2"`) in `available`.

But it'd be better to point to the archive in the `url` file instead, like this:

```
http: "https://github.com/toolslive/camltc/archive/camltc-0.9.4.tar.gz"
checksum: "9a964fdb4e68510564334fcfac6fbb62"
```

**Comment by toolslive** [comment]
- I didn't find the `opam-version` option in the manual. Will try.
- downloading the archive doesn't play well with the build as it builds
  tokyo cabinet as a git submodule.
  In the end, I guess the only thing you really want is that I cannot move the tag around,
  which would be quite hard (cryptographically) to do if I include the commit hash in the uri.

**Comment by toolslive** [comment]
`available: [ ocaml-version > "4.02.2" & opam-version > "1.2" ] `
but travis still runs the opam 1.2.0 flavour (which fails)

**Comment by yallop** [comment]
I don't know offhand how OPAM version numbers are parsed, but perhaps `opam-version > "1.2.0" ` is worth a try:

```ocaml
# "1.2.0" > "1.2";;
- : bool = true
```

**Comment by toolslive** [comment]
that works thx (maybe somebody needs to update the opam manual stating `opam-version` as available for constraints)

**Pull request merged**